Source/WebCore/Modules/webxr/WebXROpaqueFramebufferCocoa.cpp

```diff
@@ -403,16 +391,15 @@ bool WebXROpaqueFramebuffer::setupFramebuffer(GraphicsContextGL& gl, const Platf
 
     // Calculate viewports of each eye
     if (foveationChange) {
-        if (!gl.addFoveation(toIntSize(data.physicalSize[0]), toIntSize(data.physicalSize[1]), data.screenSize, data.horizontalSamples[0], data.verticalSamples, data.horizontalSamples[1]))
+        const auto& frmd = data.foveationRateMapDesc;
+        if (!gl.addFoveation(leftPhysicalSize, rightPhysicalSize, frmd.screenSize, frmd.horizontalSamples[0], frmd.verticalSamples, frmd.horizontalSamples[1]))
             return false;
         gl.enableFoveation(m_drawAttachments.colorBuffer);
     }
```

Comment:
I noticed that we call `disableFoveation()` on the graphics context always in `endFrame()`. Instead of doing it there, should we just update the foveation state here if `foveationChange` is `true`? Something like:

```
    if (foveationChange) {
        if (m_usingFoveation) {
            const auto& frmd = data.foveationRateMapDesc;
            if (!gl.addFoveation(leftPhysicalSize, rightPhysicalSize, frmd.screenSize, frmd.horizontalSamples[0], frmd.verticalSamples, frmd.horizontalSamples[1]))
                return false;
            gl.enableFoveation(m_drawAttachments.colorBuffer);
        } else
            gl.disableFoveation();
    }
```

Reply:
Foveation enable used to be global state so we wanted to disable it to avoid affecting other GL operations. Mike's changes made the map affect a texture only, so perhaps the enable/disable are redundant? (They aren't expensive)